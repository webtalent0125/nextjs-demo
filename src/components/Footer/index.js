/**
 * [Footer]
 * This component shows the use of built-in CSS support
 */

import React from 'react';

function Footer() {
  return (
    <footer className="root">
      <div className="container">
        <div className="row">
          <a
            className="link"
            href="https://nextjs.org/features/server-side-rendering"
            rel="noopener noreferrer"
            target="_blank">
            SSR
          </a>

          <a
            className="link"
            href="https://nextjs.org/features/static-exporting"
            rel="noopener noreferrer"
            target="_blank">
            Static Exporting
          </a>

          <a className="link" href="https://nextjs.org/features/css-in-js" rel="noopener noreferrer" target="_blank">
            CSS-in-JS
          </a>

          <a
            className="link"
            href="https://nextjs.org/features/progressive-web-apps"
            rel="noopener noreferrer"
            target="_blank">
            PWA
          </a>
        </div>
      </div>

      <style jsx>{`
        .root {
          bottom: 0;
          width: 100%;
          margin: 0 auto;
          padding: 0 0;
          background-image: linear-gradient(to bottom, #121212 0%, #323232 100%);
          color: #f1f1f1;
          overflow: hidden;
        }

        .container {
          margin: 0 auto;
          max-width: 1024px;
          padding: 2rem 0 4rem;
          min-height: 200px;
        }

        .row {
          display: flex;
          justify-content: space-evenly;
        }

        .link {
          text-decoration: none;
          color: #f1f1f1;
        }

        .link:hover {
          text-decoration: underline;
        }
      `}</style>
    </footer>
  );
}

export default Footer;
