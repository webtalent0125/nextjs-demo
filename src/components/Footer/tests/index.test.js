import React from 'react';

import { render } from '@testing-library/react';

import Footer from '..';

describe('<Footer />', () => {
  const spy = jest.fn();

  it('Should render correctly', () => {
    const { container } = render(<Footer t={spy} />);

    expect(container).toMatchSnapshot();
  });
});
