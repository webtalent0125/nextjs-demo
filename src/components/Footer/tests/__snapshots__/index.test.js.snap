// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Should render correctly 1`] = `
<div>
  <footer
    class="jsx-898115948 root"
  >
    <div
      class="jsx-898115948 container"
    >
      <div
        class="jsx-898115948 row"
      >
        <a
          class="jsx-898115948 link"
          href="https://nextjs.org/features/server-side-rendering"
          rel="noopener noreferrer"
          target="_blank"
        >
          SSR
        </a>
        <a
          class="jsx-898115948 link"
          href="https://nextjs.org/features/static-exporting"
          rel="noopener noreferrer"
          target="_blank"
        >
          Static Exporting
        </a>
        <a
          class="jsx-898115948 link"
          href="https://nextjs.org/features/css-in-js"
          rel="noopener noreferrer"
          target="_blank"
        >
          CSS-in-JS
        </a>
        <a
          class="jsx-898115948 link"
          href="https://nextjs.org/features/progressive-web-apps"
          rel="noopener noreferrer"
          target="_blank"
        >
          PWA
        </a>
      </div>
    </div>
  </footer>
</div>
`;
